packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -53,39 +53,88 @@ class UpdatePackagesCommand extends FlutterCommand {
+      // such package fixed at a single version across all the pubspec.yamls.
+      //
+      // First, collect up the explicit dependencies:
       final List<PubspecYaml> pubspecs = <PubspecYaml>[];
       final Map<String, PubspecDependency> dependencies = <String, PubspecDependency>{};
       final Set<String> specialDependencies = new Set<String>();
-      for (Directory directory in packages) {
+      for (Directory directory in packages) { // these are all the directories with pubspec.yamls we care about
         printTrace('Reading pubspec.yaml from: ${directory.path}');
-        final PubspecYaml pubspec = new PubspecYaml(directory);
-        pubspecs.add(pubspec);
-        for (PubspecDependency dependency in pubspec.dependencies) {
+        final PubspecYaml pubspec = new PubspecYaml(directory); // this parses the pubspec.yaml
+        pubspecs.add(pubspec); // remember it for later
+        for (PubspecDependency dependency in pubspec.dependencies) { // this is all the explicit dependencies
           if (dependencies.containsKey(dependency.name)) {
+            // If we've seen the dependency before, make sure that we are
+            // importing it the same way. There's several ways to import a
+            // dependency. Hosted (from pub via version number), by path (e.g.
+            // pointing at the version of a package we get from the Dart SDK
+            // that we download with Flutter), by SDK (e.g. the "flutter"
+            // package is explicitly from "sdk: flutter").
+            //
+            // This makes sure that we don't import a package in two different
+            // ways, e.g. by saying "sdk: flutter" in one pubspec.yaml and
+            // saying "path: ../../..." in another.
             final PubspecDependency previous = dependencies[dependency.name];
-            if (dependency.kind != previous.kind || dependency.lockTarget != previous.lockTarget)
-              throw 'Inconsistent requirements around ${dependency.name}; saw ${dependency.kind} (${dependency.lockTarget}) in "${dependency.sourcePath}" and ${previous.kind} (${previous.lockTarget}) in "${previous.sourcePath}".';
+            if (dependency.kind != previous.kind || dependency.lockTarget != previous.lockTarget) {
+              throw 'Inconsistent requirements around ${dependency.name}; '
+                    'saw ${dependency.kind} (${dependency.lockTarget}) in "${dependency.sourcePath}" '
+                    'and ${previous.kind} (${previous.lockTarget}) in "${previous.sourcePath}".';
+            }
           }
+          // Remember this dependency by name so we can look it up again.
           dependencies[dependency.name] = dependency;
+          // Normal dependencies are those we get from pub. The others we
+          // already implicitly pin since we pull down one version of the
+          // Flutter and Dart SDKs, so we track which those are here so that we
```

Comment:
grammar nit: "which those are"

Reply:
Not sure what you are asking to have changed here.